src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,171 @@
+
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public MyLinkedList() {
+        first = null;
+        last = null;
+    }
+
+    private static class Node<T> {
+        private T data;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(T data) {
+            this.data = data;
+        }
+    }
+
+    public boolean checkIndex(int index) {
+        if (index >= 0 && index < size) {
+            return true;
+        } else {
+            throw new IndexOutOfBoundsException("Wrong index");
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (isEmpty()) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+            newNode.prev = last;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode = new Node<>(value);
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            if (isEmpty()) {
+                last = newNode;
+            } else {
+                first.prev = newNode;
+                newNode.next = first;
+                first = newNode;
+                size++;
+            }
+        } else {
+            checkIndex(index);
+            Node<T> current = first;
+            for (int i = 0; i < index; i++) {
+                current = current.next;
+            }
```

Comment:
lets move logic of finding node by index to separate method (you use this logic in separate methods)